async-query/src/main/java/org/opensearch/sql/spark/transport/format/CreateAsyncQueryRequestConverter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.transport.format;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import lombok.experimental.UtilityClass;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.sql.spark.rest.model.CreateAsyncQueryRequest;
+import org.opensearch.sql.spark.rest.model.LangType;
+
+@UtilityClass
+public class CreateAsyncQueryRequestConverter {
```

Comment:
Can you see this commit: https://github.com/vamsi-amazon/sql/commit/fd75fcd7949240a31c18469b429ccbb2e32ba3e4#diff-b8da527871e1adbe7ec8f4e08dd75bf5b426be4d195c9b0860e09260c5dda127
Also, is this the only model left to have dependency on XContent?
Are there other models in async-query-core which have dependency on XContent.

Reply:
Yes, other XContent dependent classes are already in `async-query` module since they are OpenSearch dependent implementation.